pootle/core/views.py

```diff
@@ -92,6 +92,9 @@ def class_wrapper(f):
         def method_wrapper(self, request, *args, **kwargs):
             directory_permission = check_directory_permission(
                 permission, request, self.permission_context)
+            if directory_permission and self.required_permission:
```

Comment:
Why is it necessary to check for `directory_permission` if it is being overwritten?